rclpy/src/rclpy/context.hpp

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLPY__CONTEXT_HPP_
+#define RCLPY__CONTEXT_HPP_
+
+#include <Python.h>
+#include <pybind11/pybind11.h>
+#include <rcl/context.h>
+#include <rcl/types.h>
+
+#include <stdexcept>
+
+extern "C" {
+#include "rclpy_common/handle.h"
+}
+
+#include "rclpy_common/exceptions.hpp"
+
+namespace py = pybind11;
+
+namespace rclpy
+{
+/// Retrieves domain id from init_options of context
+/**
+ * \param[in] pycontext Capsule containing rcl_context_t
+ * \return domain id
+ */
+size_t
+context_get_domain_id(py::capsule pycontext)
```

Comment:
Is the idea to only ever include this file once? I'm worried about symbol collisions again.

Reply:
Split `context.hpp` into `.cpp` and `.hpp` in 4f5c39a7fa87071cd52c2418f5bb520a78be1529.

I would expect this to only be included once in the first pass, but once we convert to a `Context` class then I would expect it to be included multiple times.